### PR TITLE
Remove cloud specific DNS check to make e2e Conformance eligible

### PR DESCRIPTION
**Pull request by mgdevstack**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Following changes done to make e2e Conformance promotable -
1. Removed google cloud specific DNS resolution check, as another Conformance E2E confirms that it is resolvable. i.e [effective e2e](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/dns.go#L52-L60): `should provide DNS for the cluster`.

2. Removed `framework.NodeOSDistroIs("windows")` block as it's not needed for [LinuxOnly] tagged e2e.

**Which issue(s) this PR fixes**:
This helps E2E `should resolve DNS of partial qualified names for the cluster [LinuxOnly]` to get promoted in #74977 .

Fixes #

**Special notes for your reviewer**:
This is blocking #74977 to get merged.

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/area conformance
/sig testing
@kubernetes/cncf-conformance-wg

### CODE REVIEW

**Comment by cmluciano** [comment]
This seems benign to me but I'll defer to John since he's been more involved with this recently

/assign @johnbelamaric

**Review by cmluciano — COMMENTED** [review]
This seems benign to me but I'll defer to John since he's been more involved with this recently

/assign @johnbelamaric

File: test/e2e/network/dns.go
Comment (spiffxp) [inline test/e2e/network/dns.go]: I would rather see this block broken out into its own testcase, same for the conformance test above

these names are important to verify, they're just not intended to be exercised as part of a conformance test

**Review by spiffxp — CHANGES_REQUESTED** [review]
I don't think this quite addresses the ask we had, ref: https://github.com/kubernetes/kubernetes/pull/74977#issuecomment-515531666

>Create a new PR that removes Windows check from LinuxOnly tests, and moves GCE-specific tests to a new test.

eg:, at the end of this I would expect:
- `should provide DNS for the cluster` - kubernetes.default.svc.(clusterdnsname) - conformance
- `should provide DNS for the cluster [Provider:GCE]` - google.com, metadata - regular e2e, no NodeOSDistroIs check
- `should resolve DNS of partial qualified names for the cluster [LinuxOnly]` - kubernetes.default, kubernetes.default.svc - regular e2e

File: test/e2e/network/dns.go
Comment (mgdevstack) [reply test/e2e/network/dns.go]: Done.

File: test/e2e/network/dns.go
Comment (johnbelamaric) [inline test/e2e/network/dns.go]: @spiffxp  is `[Provider:GCE]` equivalent to `framework.ProviderIs("gce", "gke")` in the e2e framework?

Comment (spiffxp) [reply test/e2e/network/dns.go]: I completely made the tag up out of whole cloth but I feel like we're going to want some way to signify that we want these provider-specific tests to run.  I could do something like `[Feature:GCEProvider]` as well, but this would exclude it from all of the release-master-blocking tests, almost all of which run in GCE and need this tested anyway.  Open to feedback on a more appropriate way to do things here

**Comment by spiffxp** [comment]
/lgtm
/hold
/assign @bgrant0607 @smarterclayton 
for approval by conformance approvers

I would approve on my own but want some feedback on whether the tag I proposed is appropriate

File: test/e2e/network/dns.go
Comment (johnbelamaric) [reply test/e2e/network/dns.go]: I don't see how this can go up for approval right now, this new test will fail on other providers. I think we need to add a skip in the new test if it's not GCE or GKE. I don't know that we need to decide on a label for it right now.

Comment (spiffxp) [reply test/e2e/network/dns.go]: This test isn't intended to be a Conformance test though, the Conformance part of it will still pass just fine.  This was in service of allowing other tests to be promoted to Conformance (see https://github.com/kubernetes/kubernetes/pull/74977)

Comment (johnbelamaric) [reply test/e2e/network/dns.go]: But why are we adding a failing e2e test (regardless of any conformance question)?

Comment (bgrant0607) [reply test/e2e/network/dns.go]: Once we decide on the best way to do this, please update the documentation
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md#kinds-of-tests

**Comment by mgdevstack** [comment]
Pinging to get this merged as it's blocking #74977 for 1.16 release cycle.

**Comment by spiffxp** [comment]
/lgtm cancel
discussed during conformance meeting today
@johnbelamaric's point in https://github.com/kubernetes/kubernetes/pull/80682#discussion_r309370687 was that the gce-specific test should _also_ have a `Skip` call in it

**Comment by mgdevstack** [comment]
Added `framework.SkipUnlessProviderIs("gce", "gke")` for `[Provider:GCE]` e2e.

File: test/e2e/network/dns.go
Comment (mgdevstack) [reply test/e2e/network/dns.go]: https://github.com/kubernetes/kubernetes/pull/80682#discussion_r309370687 Addressed.

https://github.com/kubernetes/kubernetes/blob/20010af0de0184410107fb4c7d2f0c4d74db5f15/test/e2e/network/dns.go#L65

**Comment by mgdevstack** [comment]
Hi @johnbelamaric, @spiffxp 
Is this PR good for approval now?

File: test/e2e/network/dns.go
Comment (johnbelamaric) [inline test/e2e/network/dns.go]: metadata should only be appended if it's not windows, according to previous code

Comment (mgdevstack) [reply test/e2e/network/dns.go]: Done.

**Pull request merged**
